### PR TITLE
Extract in a method vals generation of mo lines for easier inherit

### PR DESCRIPTION
**Pull request by Viggor**
Put in a method the generation of MO conssumed products lines for easier inherit in modules.

### CODE REVIEW

File: addons/mrp/mrp.py [hunk 1]
```diff
@@ -264,6 +264,17 @@ def _skip_bom_line(self, cr, uid, line, product, context=None):
                 return True
         return False
 
+    def _prepare_conssumed_line(self, cr, uid, bom_line_id, quantity, product_uos_qty, context=None):
```
Comment (sebastienbeau) [inline addons/mrp/mrp.py]: can you rename bom_line_id into bom_line ? Indeed it's not a id even if in the original code then call it bom_line_id I think it's a bad idea to keep this name here

File: addons/mrp/mrp.py [hunk 2]
```diff
@@ -322,14 +333,8 @@ def _factor(factor, product_efficiency, product_rounding):
...
             if bom_line_id.type != "phantom" and (not bom_id or self.browse(cr, uid, bom_id, context=context).type != "phantom"):
-                result.append({
-                    'name': bom_line_id.product_id.name,
-                    'product_id': bom_line_id.product_id.id,
-                    'product_qty': quantity,
-                    'product_uom': bom_line_id.product_uom.id,
-                    'product_uos_qty': bom_line_id.product_uos and _factor(bom_line_id.product_uos_qty * factor, bom_line_id.product_efficiency, bom_line_id.product_rounding) or False,
-                    'product_uos': bom_line_id.product_uos and bom_line_id.product_uos.id or False,
-                })
+                product_uos_qty = bom_line_id.product_uos and _factor(bom_line_id.product_uos_qty * factor, bom_line_id.product_efficiency, bom_line_id.product_rounding) or False
```
Comment (sebastienbeau) [inline addons/mrp/mrp.py]: I understand that the method _factor is only accessible in this method as it's declared here. This forbid you to use it in the _prepare method. Maybe the best solution is to declare this method for the full file at the top of mrp.py before declaring the first class? so you can use it in the prepare method

**Comment by sebastienbeau** [comment]
This change is a must to have to extend the mrp in 8 version, it do not change the current api so after the small point I ask I really think we should merged it

File: addons/mrp/mrp.py (hunk 1 above)
Comment (Viggor) [inline addons/mrp/mrp.py]: Done!

File: addons/mrp/mrp.py (hunk 2 above)
Comment (Viggor) [inline addons/mrp/mrp.py]: I move factor method outside of the method so i can delete the paramater product_uos_qty which is computed inside the new method but for the quantity parameter it is used after this new method so it still be passed.

**Comment by bealdav** [comment]
LGTM for me :+1:

**Comment by mart-e** [comment]
Hello,

Thank you for your contribution but the version 8.0 is no longer supported. We only support the last 3 stable versions so no longer accepts patches into this branch.
We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.
Regards

**Pull request closed without merging**